src/knownhost.c

```diff
@@ -421,19 +421,22 @@ knownhost_check(LIBSSH2_KNOWNHOSTS *hosts,
                     */
                     unsigned char hash[SHA_DIGEST_LENGTH];
                     libssh2_hmac_ctx ctx;
-                    libssh2_hmac_ctx_init(ctx);
+                    _libssh2_hmac_ctx_init(&ctx);
```

Comment:
Given that this function has just changd to return int instead of void, should we check the return value here?

Reply:
Thank you, good spot, we should indeed! I'v pushed an update.